### ISSUE TITLE
Shadowing not detected in comprehensions within comprehensions

### ISSUE THREAD

**Issue by Stebalien** [PyCQA/pylint#982]
Pylint correctly detects variable shadowing in 1-deep list comprehensions but fails in list comprehensions within list comprehensions.

The following works without warning:

``` python
def test():
    var = 1
    [var["a"] for var in []]
```

However, the following:

``` python
def test():
    var = 1
    #  ↓↓↓
    [[ var["a"] for i in range(0, 10)] for var in []]
```

Warns that `Value 'var' is unsubscriptable (unsubscriptable-object)`.
### pylint --version output

```
pylint 1.5.6, 
astroid 1.4.6
Python 3.5.2 (default, Jun 28 2016, 08:46:01) 
[GCC 6.1.1 20160602]
```

**Comment by PCManticore** [comment]
Thanks.

**Comment by mbyrnepr2** [comment]
Neither example emits `unsubscriptable-object` on the latest pylint in my tests. @Pierre-Sassoulas  can this be closed?

**Comment by Pierre-Sassoulas** [comment]
Thank you for triaging @mbyrnepr2, much appreciated !
